unitTests/ui/QuerySuggestPreviewTest.ts

```diff
@@ -0,0 +1,237 @@
+    let suggestionManager: SuggestionsManager;
+    let suggestion: Dom;
+    let elementInsideSuggestion: Dom;
+
+    function setupQuerySuggestPreview(options: IQuerySuggestPreview = {}) {
+      const tmpl: HtmlTemplate = Mock.mock<HtmlTemplate>(HtmlTemplate);
+      options['resultTemplate'] = tmpl;
+
+      test = Mock.advancedComponentSetup<QuerySuggestPreview>(
+        QuerySuggestPreview,
+        new Mock.AdvancedComponentSetupOptions(null, options, env => testEnv)
+      );
+    }
+
+    function triggerQuerySuggestHover(suggestion: string = 'test') {
+      (test.env.searchEndpoint.search as jasmine.Spy).and.returnValue(
+        Promise.resolve(FakeResults.createFakeResults(test.cmp.options.numberOfPreviewResults))
+      );
+      $$(testEnv.root).trigger(OmniboxEvents.querySuggestGetFocus, { suggestion });
+    }
+
+    function buildSuggestion() {
+      container = $$(document.createElement('div'));
+      suggestionContainer = $$(document.createElement('div'));
+      suggestion = $$(document.createElement('div'));
+      elementInsideSuggestion = $$(document.createElement('div'));
+
+      suggestion.el.appendChild(elementInsideSuggestion.el);
+      suggestionContainer.el.appendChild(suggestion.el);
+      container.el.appendChild(suggestionContainer.el);
+    }
+
+    function setupSuggestionManager() {
+      buildSuggestion();
+      const inputManager = new InputManager(document.createElement('div'), () => {}, {} as MagicBoxInstance);
+
+      suggestionManager = new SuggestionsManager(suggestionContainer.el, document.createElement('div'), testEnv.root, inputManager);
+    }
+
+    function setupRenderPreview() {
+      test.env.root.appendChild(suggestionContainer.el);
+      (test.cmp.options.resultTemplate.instantiateToElement as jasmine.Spy).and.returnValue(Promise.resolve($$('div').el));
+    }
+
+    function setupSuggestion(suggestions: Suggestion[] = [{ text: 'test' }]) {
+      setupSuggestionManager();
+      suggestionManager.updateSuggestions(suggestions);
+      setupRenderPreview();
+    }
+
+    beforeEach(() => {
+      testEnv = new Mock.MockEnvironmentBuilder();
+    });
+
+    describe('expose options', () => {
+      it('numberOfPreviewResults set the number of results to query', done => {
+        const numberOfPreviewResults = 5;
+        setupQuerySuggestPreview({ numberOfPreviewResults });
+        setupSuggestion();
+        triggerQuerySuggestHover();
+        setTimeout(() => {
+          expect(test.cmp.queryController.getLastQuery().numberOfResults).toBe(numberOfPreviewResults);
+          done();
+        }, test.cmp.options.hoverTime);
+      });
+
+      it('hoverTime set the time before the query is executed', done => {
+        const hoverTime = 200;
+        setupQuerySuggestPreview({ hoverTime });
+        setupSuggestion();
+        triggerQuerySuggestHover();
+        expect(test.cmp.queryController.getLastQuery).not.toHaveBeenCalled();
+        setTimeout(() => {
```

Comment:
Could use jasmine.clock() for all those setTimeout for all those tests.

Reply:
I will look into it